spec/features/sessions/user_sign_up_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'devise/registrations/new.html.erb', type: :feature do
+  before do
+    visit new_user_registration_path
+  end
+
+  it 'has input fields for first name and list name' do
+    expect(page).to have_field 'user[user_profile_attributes][first_name]'
+    expect(page).to have_field 'user[user_profile_attributes][last_name]'
+  end
+
+  it 'creates a user with its profile data' do
+    page.fill_in 'user[user_profile_attributes][first_name]', with: 'Max'
+    page.fill_in 'user[user_profile_attributes][last_name]', with: 'Mustermann'
+
+    page.fill_in 'user[email]', with: 'max@mustermann.com'
+    page.fill_in 'user[password]', with: '123456'
+    page.fill_in 'user[password_confirmation]', with: '123456'
+
+    page.find('input[type=submit]').click
```

Comment:
This submit button will cause the `app/controllers/application_controller.rb` to run
 `def after_sign_in_path_for(_resource)
    vm_index_path
  end`
which then leads to VM index page which tries to collect VM information via the `app/api/vmapi.rb` which will not be successfull out side of HPI network.

Reply:
This should be fine now, as it's not intended behaviour for the index page to throw an error when trying to load it.